phases/19-capstone-projects/81-end-to-end-distributed-train/tests/test_e2e.py

```
@@ -0,0 +1,75 @@
+        self.assertEqual(len(losses), 20)
+
+    def test_manifest_present_at_checkpoint_step(self):
+        loaded = load_sharded(self.ckpt_dir, expected_world_size=WORLD_SIZE)
+        self.assertEqual(len(loaded), WORLD_SIZE)
+        for r in range(WORLD_SIZE):
+            self.assertIn("model_state", loaded[r])
+            self.assertIn("optim_state", loaded[r])
+
+    def test_loss_finite_and_no_nan(self):
+        losses = self.results[0]["losses"]
+        for s, l in enumerate(losses):
```

Ruff (0.15.14)

[error] Ambiguous variable name: `l`

(E741)